test/support.py

```diff
@@ -26,6 +26,12 @@ class DefaultExtensionTemplate(object):
         .defines = moduledefs
     };
 
+    #define MODINIT_ERROR        \
+        do {                     \
+            HPy_Close(ctx, m);   \
+            return HPy_NULL;     \
+        } while (0)
+
```

Comment:
is there any reason to use a macro instead of a `static inline` function? E.g.:
```c
HPy _ModInit_Error(HPyContext ctx, HPy m) {
    HPy_Close(ctx, m);
    return HPy_NULL;
}
```

Reply:
Using `#define` means that the caller doesn't have to know what needs to be cleaned up. E.g. `_ModInit_Error(ctx, m)` is the right signature now because `m` is the only thing that needs to be cleaned up, but it could be the wrong signature later if more needs to be cleaned up.

If you feel strongly I'm happy to use the static inline now and then just fix all the code later if we ever need to clean up more things.

Reply:
I don't feel strongly so I'm fine with either solution: this is something which is just used by our own tests anyway.
It's just that it's too easy to introduce nasty bugs, see e.g.  bca2441. Also, `MODINIT_ERROR` implicitly assumes that there is a `ctx` variable instead of getting one "properly", which looks wrong.

Reply:
I converted these to a `goto MODINIT_ERROR` which is what I wanted to do originally, but I was worried that the compiler would complain about unreachable code when it wasn't used. So I made sure it used. :)